include/bluetooth/services/ancs_c.h

```diff
@@ -0,0 +1,580 @@
+	/** Identifies that the attribute data is a "Subtitle". */
+	BT_GATT_ANCS_NOTIF_ATTR_ID_SUBTITLE,
+	/** Identifies that the attribute data is a "Message". */
+	BT_GATT_ANCS_NOTIF_ATTR_ID_MESSAGE,
+	/** Identifies that the attribute data is a "Message Size". */
+	BT_GATT_ANCS_NOTIF_ATTR_ID_MESSAGE_SIZE,
+	/** Identifies that the attribute data is a "Date". */
+	BT_GATT_ANCS_NOTIF_ATTR_ID_DATE,
+	/** The notification has a "Positive action" that can be executed associated with it. */
+	BT_GATT_ANCS_NOTIF_ATTR_ID_POSITIVE_ACTION_LABEL,
+	/** The notification has a "Negative action" that can be executed associated with it. */
+	BT_GATT_ANCS_NOTIF_ATTR_ID_NEGATIVE_ACTION_LABEL
+} bt_gatt_ancs_c_notif_attr_id_val_t;
+
+/**@brief Flags for iOS notifications. */
+typedef struct {
+	/** If this flag is set, the notification has a low priority. */
+	uint8_t silent : 1;
+	/** If this flag is set, the notification has a high priority. */
+	uint8_t important : 1;
+	/** If this flag is set, the notification is pre-existing. */
+	uint8_t pre_existing : 1;
+	/** If this flag is set, the notification has a positive action that can be taken. */
+	uint8_t positive_action : 1;
+	/** If this flag is set, the notification has a negative action that can be taken. */
+	uint8_t negative_action : 1;
+} bt_gatt_ancs_c_notif_flags_t;
+
+/**@brief Parsing states for received iOS notification and app attributes. */
+typedef enum {
+	/** Parsing the command ID. */
+	COMMAND_ID,
+	/** Parsing the notification UID. */
+	NOTIF_UID,
+	/** Parsing app ID. */
+	APP_ID,
+	/** Parsing attribute ID. */
+	ATTR_ID,
+	/** Parsing the LSB of the attribute length. */
+	ATTR_LEN1,
+	/** Parsing the MSB of the attribute length. */
+	ATTR_LEN2,
+	/** Parsing the attribute data. */
+	ATTR_DATA,
+	/** Parsing is skipped for the rest of an attribute (or entire attribute). */
+	ATTR_SKIP,
+	/** Parsing for one attribute is done. */
+	DONE
+} bt_gatt_ancs_c_parse_state_t;
+
+/**@brief iOS notification structure. */
+typedef struct {
+	/** Notification UID. */
+	uint32_t notif_uid;
+	/** Whether the notification was added, removed, or modified. */
+	bt_gatt_ancs_c_evt_id_values_t evt_id;
+	/** Bitmask to signal whether a special condition applies to the notification. For example, "Silent" or "Important". */
+	bt_gatt_ancs_c_notif_flags_t evt_flags;
+	/** Classification of the notification type. For example, email or location. */
+	bt_gatt_ancs_c_category_id_val_t category_id;
+	/** Current number of active notifications for this category ID. */
+	uint8_t category_count;
+} bt_gatt_ancs_c_evt_notif_t;
+
+/**@brief iOS attribute structure. This type is used for both notification attributes and app attributes. */
+typedef struct {
+	/** Length of the received attribute data. */
+	uint16_t attr_len;
+	/** Classification of the attribute type. For example, "Title" or "Date". */
+	uint32_t attr_id;
+	/** Pointer to where the memory is allocated for storing incoming attributes. */
+	uint8_t *p_attr_data;
```

Comment:
Prefix `p_` should not be used in NCS, we don't follow that naming convention.

Reply:
Done